docs/architecture.md

```diff
@@ -1,188 +1,260 @@
+The Kubewarden stack consists of these components:
 
-- Kubewarden Custom Resources: these are [Kubernetes Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
-  that simplify the process of managing policies.
+- Kubewarden Custom Resources:
+These are [Kubernetes Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+that simplify the process of managing policies.
 
 - [`kubewarden-controller`](https://github.com/kubewarden/kubewarden-controller):
-  this is a Kubernetes controller that reconciles Kubewarden's Custom Resources.
-  This component creates parts of the Kubewarden stack and, most important of
-  all, translates Kubewarden's concepts into native Kubernetes directives.
+This is a Kubernetes controller that reconciles Kubewarden's Custom Resources.
+This controller creates parts of the Kubewarden stack.
+It also translates Kubewarden configuration into Kubernetes directives.
 
-- Kubewarden policies: these are WebAssembly modules that hold the validation
-  or mutation logic. These are covered in depth inside of [this chapter](/writing-policies/index.md).
+- Kubewarden policies:
+These are WebAssembly modules holding the validation or mutation logic.
+WebAssembly modules have detailed documentation in the
+[writing policies](/writing-policies/index.md) sections.
 
 - [`policy-server`](https://github.com/kubewarden/policy-server):
-  this component receives the requests to be validated. It does that
-  by executing Kubewarden's policies.
+The `policy-server` receives requests for validation.
+It validates the requests by executing Kubewarden policies.
 
-At the bottom of the stack, Kubewarden's integrates with Kubernetes using the
-concept of [Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+Kubewarden integrates with Kubernetes using
+[Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
 In particular, Kubewarden operates as a Kubernetes Admission Webhook.
-`policy-server` is the actual Webhook endpoint that is reached by Kubernetes
-API server to validate relevant requests.
+The `policy-server` is the Webhook endpoint called by the Kubernetes API server to validate requests.
 
-Kubernetes is made aware of the existence of Kubewarden's Webhook endpoints by
-`kubewarden-controller`. This is done by registering either
-a `MutatingWebhookConfiguration` or a `ValidatingWebhookConfiguration`
-object.
+<!--TODO:Have I got this next paragraph right?-->
 
-This diagram shows the full architecture overview of a cluster running
-the Kubewarden stack:
+Kubernetes discovers Kubewarden's Webhook endpoints using `kubewarden-controller`.
+This works by Kubewarden registering either a `MutatingWebhookConfiguration`
+or a `ValidatingWebhookConfiguration` object with Kubernetes.
```

Comment:
Agree with José. Instead of "Kubernetes discovers Kubewarden's webhooks", it should be directly "`kubewarden-controller` registers the needed `MutatingWebhookConfiguration`
or a `ValidatingWebhookConfiguration` objects in the Kubernetes apiserver".

Reply:
Thanks, Changed as per @viccuad suggestion.